core/src/test/java/io/aiven/klaw/service/TopicSyncControllerServiceTest.java

```diff
@@ -177,6 +188,198 @@ public void getSyncTopics() throws Exception {
+
+    // execute
+    ApiResponse retval =
+        topicSyncControllerService.updateSyncBackTopics(
+            createSyncBackTopic(ALL_TOPICS, new String[0]));
+
+    // verfiy only 1 topic Request is sent to the manage database as the other already exists.
+    verify(handleDbRequests).requestForTopic(topicRequestCaptor.capture());
+    verify(handleDbRequests).updateTopicRequest(updateTopicRequestCaptor.capture(), eq(USERNAME));
+
+    List<TopicRequest> req = topicRequestCaptor.getAllValues();
+    List<TopicRequest> update = updateTopicRequestCaptor.getAllValues();
+    // we have two topics but one already exists so only one should be added to the database.
+    assertEquals(1, req.size());
+    assertEquals(1, update.size());
+    // we expect all request to also be updated to approved
+    assertEquals(req.size(), update.size());
+
+    verifyCaptureContents(req, update, 0, 1, "UnitTest-01");
+
+    // This should pass when clusterApi is updated
+    assertNotEquals(
+        "Error :Could not approve topic request. Please contact Administrator.",
+        retval.getResult());
+    assertEquals("success", retval.getResult());
+  }
+
+  @Test
+  @Order(6)
+  public void approveTopicRequestAllTopicsCreateAll() throws KlawException {
+
+    stubUserInfo();
+    MockMultipleTopics();
+    mockGetTopicsFromEnv();
+    when(clusterApiService.approveTopicRequests(
+            any(), anyString(), anyInt(), anyString(), anyString(), any(), eq(TENANT_ID)))
+        .thenReturn(createAPIResponse(HttpStatus.OK, ApiResultStatus.SUCCESS.value));
+
+    // execute
+    ApiResponse retval =
+        topicSyncControllerService.updateSyncBackTopics(
+            createSyncBackTopic(ALL_TOPICS, new String[0]));
+
+    // verfiy exactly 2 topics Requests are sent to the manage database
+    verify(handleDbRequests, times(2)).requestForTopic(topicRequestCaptor.capture());
+    verify(handleDbRequests, times(2))
+        .updateTopicRequest(updateTopicRequestCaptor.capture(), eq(USERNAME));
+
+    List<TopicRequest> req = topicRequestCaptor.getAllValues();
+    List<TopicRequest> update = updateTopicRequestCaptor.getAllValues();
+    // we have two new topics both should be added to the database.
+    assertEquals(2, req.size());
+    assertEquals(2, update.size());
+    // we expect all request to also be updated to approved
+    verifyCaptureContents(req, update, 0, 1, "UnitTest-01");
+
+    verifyCaptureContents(req, update, 1, 2, "UnitTest-02");
+    assertNotEquals(
+        "Error :Could not approve topic request. Please contact Administrator.",
+        retval.getResult());
+    assertEquals("success", retval.getResult());
+  }
+
+  @Test
+  @Order(7)
+  public void approveTopicRequestSelectedWhereOneTopicIsAlreadyCreated() throws KlawException {
+    stubUserInfo();
+    MockMultipleTopics();
+    mockSelectedOnlyTopics(1, "UnitTest-01", env.getId());
```

Comment:
Is it better to declare static variables for these 2 topics 01 and 02?  Seems like they are referred in multiple places

Reply:
Yes I agree, I will create a static class variable.

Reply:
As discussed a constant has been created for these topic names for the unit tests.
  public static final String TOPIC_NAME_1 = "unittest-01";
  public static final String TOPIC_NAME_2 = "unittest-02";